python/GafferUI/NodeSetEditor.py

```diff
@@ -61,22 +77,149 @@ def __init__( self, topLevelWidget, scriptNode, **kw ) :
+	# As members are added to and removed from the set, the UI will be updated
+	# automatically to show them. This also calls `nodeSet.setRemoveOrphans( True
+	# )` so that deleted nodes are not visible in the UI.
+	#
+	# This will break any editor links, if set.
+	# NOTE: The driver will be updated *after* the node set, such that calling
+	# `getNodeSetDriver` in the nodeSetChangedSignal will return the departing
+	# driver. If the listener requires accurate information about the driver, then
+	# the nodeSetDriverChanegd signal should be used instead.
 	def setNodeSet( self, nodeSet ) :
 
 		self.__setNodeSetInternal( nodeSet, callUpdateFromSet=True )
+		# We do this after setting the node set, so that when the driver changed
+		# signal is emitted, we will have the new node set. Otherwise the editor
+		# looks like it has the old drivers node set still despite not having a
+		# driver...
+		self.setNodeSetDriver( None )
 
 	def getNodeSet( self ) :
 
 		return self.__nodeSet
 
+	## Called before nodeSetDriverChangedSignal in the event that setNodeSet breaks a driver link.
 	def nodeSetChangedSignal( self ) :
 
 		return self.__nodeSetChangedSignal
 
+	## Links the nodeSet for this editor to that of the supplied drivingEditor.
+	# The default mode results in a simple mirroring of the driver's node set
+	# to this editor. Other modes may be registered by other Gaffer modules.
+	# If drivingEditor is None, any existing links will be broken.
+	def setNodeSetDriver( self, drivingEditor, mode = DriverModeNodeSet ) :
+
+		if drivingEditor is not None :
+			assert( isinstance( drivingEditor, GafferUI.NodeSetEditor ) )
+			# We also need to stop people creating infinite loops
+			if GafferUI.NodeSetEditorAlgo.drivesNodeSet( self, drivingEditor ) :
+				raise ValueError( "The supplied driver is already driven by this editor" )
+
+		if self.__nodeSetDriver :
+			previousDriver = self.__nodeSetDriver["weakDriver"]()
+			# It may have died, so we sill will have data, but the ref wil be dead
+			if previousDriver is not None :
+				if drivingEditor is previousDriver and mode == self.__nodeSetDriver["mode"] :
+					return
+				else :
+					previousDriver.__unregisterDrivenEditor( self )
+
+		self.__nodeSetDriver = {}
+
+		if drivingEditor :
+
+			drivingEditor.__registerDrivenEditor( self, mode )
+
+			weakDriver = weakref.ref(
+				drivingEditor,
+				# We need to unlink ourselves if the driver goes away
+				lambda _ : self.setNodeSetDriver( None )
+			)
+			changeCallback = self.__nodeSetDriverModes[ mode ]
+
+			def updateFromDriver( _ ) :
+				if weakDriver() is not None :
+					nodeSet = weakDriver().getNodeSet()
+					if changeCallback :
+						nodeSet = changeCallback( self, weakDriver() )
+					# We need to be able to track un-pinning, so we need to see if someone
+					# else has changed the nodeSet of the editor by some other means, so
+					# keep a copy here.
+					self.__nodeSetDriver["lastNodeSet"] = nodeSet
```

Comment:
I can't find any other reference to this - is it doing anything?

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/3270/commits/d0aa0e2d70aaae24dd0d459b5fc99220291c8da7